delete-unused-fields/src/callApi.js

```diff
@@ -0,0 +1,53 @@
+module.exports = function callApi(settings, cb) {
+    const h = require('https');
+    const port = 443;
+
+    let path = '';
+    let shortUrlWithPath = settings.url.substring(8);
```

Comment:
I find all these substrings hard to follow. 
And you don't need it, you can initialise your options with URL()

try this:
```javascript
const h = require('https');

let options = new URL(settings.url);

// set method and headers if passed in
['method', 'headers'].forEach(attr=>{
  if (settings[attr]) {
     options[attr]= settings[attr];
  }
});

const req = h.request(options, (res) => {
...
```

Reply:
or a shorter version:
```javascript
const h = require('https');

let options = Object.assign({}, new URL(settings.url), settings));

const req = h.request(options, (res) => {
...
```